File: week-02/day-03/string02.py

# Create a function called 'reverse_string' which takes a string as a parameter
# The function reverses it and returns with the reversed string

#Solution 1
reversed = ".eslaf eb t'ndluow ecnetnes siht ,dehctiws erew eslaf dna eurt fo sgninaem eht fI"
reversed = reversed[::-1]
print(reversed)

#Solution 2
def slowreverse(text):
    s = ''
    for i in text:
        s = i + s
    print (s)
slowreverse (".eslaf eb t'ndluow ecnetnes siht ,dehctiws erew eslaf dna eurt fo sgninaem eht fI")

#Solution 3
reversed = ".eslaf eb t'ndluow ecnetnes siht ,dehctiws erew eslaf dna eurt fo sgninaem eht fI"
def reverse(x):
    a = x[::-1]
    print(a)
reverse(reversed)

File: week-07/day-02/string02.js
'use strict';
// Reverse the string!
// var reversed = ".eslaf eb t'ndluow ecnetnes siht ,dehctiws erew eslaf dna eurt fo sgninaem eht fI"


// Solution 1
var reversed = ".eslaf eb t'ndluow ecnetnes siht ,dehctiws erew eslaf dna eurt fo sgninaem eht fI"
reversed = reversed.split("").reverse().join("");
console.log(reversed);

// Solution 2
var reversed = ".eslaf eb t'ndluow ecnetnes siht ,dehctiws erew eslaf dna eurt fo sgninaem eht fI"

function reverse(string) {
    for (var i = string.length - 1, newString = ''; i >= 0; newString += string[i--]) { }
    return newString;
  }
console.log(reverse(reversed))

// Solution 3
var reversed = ".eslaf eb t'ndluow ecnetnes siht ,dehctiws erew eslaf dna eurt fo sgninaem eht fI"
function reverse(string) {
    var newString = '';
    for (var i = string.length - 1; i >= 0; i--)
    newString += string[i];
    return newString;
  }
console.log(reverse(reversed))

